### PR TITLE
WPP-61 Adds pre-interceptor for user attributes in header values

### PR DESCRIPTION
**Pull request by vertein**
You can configure your portlet to use user-attributes and pass them along in headers of your own naming.

Also configured javadoc plugin to work with those still using java 7 (I know, not using for long).

### CODE REVIEW

File: docs/wpp_config.md [hunk 1]
```diff
@@ -107,6 +107,57 @@ Note: the keys need to be defined as user attributes in the portlet definition f
...
+  <value>displayName</value>
+  <value>cn</value>
+  <value>givenName</value>
+  <value>mail</value>
+  <value>telephoneNumber</value>
+</portlet-preference>
+```
+
+Those are the names of the headers you will be sending. The user attributes you
+will send are  in the porlte preference named
```
Comment (timlevett) [inline docs/wpp_config.md]: portlet?

File: src/test/java/org/jasig/portlet/proxy/service/web/interceptor/HeaderPassingPreInterceptorTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  HeaderPassingPreInterceptor preprocessor;
+  @Mock PortletRequest portletRequest;
+  HttpContentRequestImpl proxyRequest;
+  Map<String, String> userInfoMap;
+  @Mock PortletPreferences prefs;
+  String[] headerNames;
+  String[] headerValues;
+
+  @Before
+  public void setUp() {
```
Comment (timlevett) [inline src/test/java/org/jasig/portlet/proxy/service/web/interceptor/HeaderPassingPreInterceptorTest.java]: yay for testing

File: docs/wpp_config.md (hunk 1 above)
Comment (vertein) [inline docs/wpp_config.md]: Fixed spelling mistake.

**Comment by bjagg** [comment]
Great stuff! Thank you for maintaining the new docs

**Comment by bjagg** [comment]
https://issues.jasig.org/browse/WPP-61

**Pull request merged**